vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNavItem.java

```diff
@@ -280,11 +301,58 @@ public String getPath() {
      */
     public void setQueryParameters(QueryParameters queryParameters) {
         this.queryParameters = queryParameters;
-        refresh();
+        // Apply new query parameters to the path
+        setPath(getPath());
     }
 
-    private void refresh() {
-        setPath(getPath());
+    /**
+     * Gets the path aliases for this item.
+     *
+     * @return the path aliases for this item, empty if none
+     */
+    public Set<String> getPathAliases() {
+        Serializable pathAliasesRaw = getElement()
+                .getPropertyRaw("pathAliases");
+        if (pathAliasesRaw == null) {
+            return Collections.emptySet();
+        }
+        return new HashSet<>(JsonSerializer.toObjects(String.class,
+                (JsonArray) pathAliasesRaw));
+    }
+
+    /**
+     * Sets the specified path aliases to this item. The aliases act as
+     * secondary paths when determining the active state of the item.
+     *
+     * @param pathAliases
+     *            the path aliases to set to this item
+     */
+    public void setPathAliases(Set<String> pathAliases) {
+        if (pathAliases.isEmpty()) {
```

Comment:
`null` also can be set

Reply:
Should we allow setting null? An empty set already signifies the same thing. Maybe we can put an exception clause like with the others.

Reply:
I assumed that `null` can be set because it just seems to be a common way of resetting a value in our code base.

Reply:
I checked 2 examples of `setValue(Set)` from `CheckboxGroup` and `MultiselectComboBox`. The first accepts null, while it throws and exception for the latter. We can go for the first approach, and also mention in in the JavaDoc.